### PR TITLE
Removes circular dependency of secondary toolbar on app.js.

### LINKED ISSUE #7204: Remove the circular dependency hack for the secondary toolbar
We should remove the hack at https://github.com/mozilla/pdf.js/blob/master/web/secondary_toolbar.js#L177.

### PR DESCRIPTION
**Pull request by yurydelendik**
Fixes #7204

### CODE REVIEW

File: web/app.js [hunk 1]
```diff
@@ -1937,11 +1984,16 @@ window.addEventListener('DOMMouseScroll', handleMouseWheel);
...
 
 window.addEventListener('click', function click(evt) {
-  if (SecondaryToolbar.opened &&
-      PDFViewerApplication.pdfViewer.containsElement(evt.target)) {
+  if (!SecondaryToolbar.opened) {
+    return;
+  }
+  var appConfig = PDFViewerApplication.appConfig;
+  if (PDFViewerApplication.pdfViewer.containsElement(evt.target) ||
+      (appConfig.toolbar.container.contains(evt.target) &&
```
Comment (Snuffleupagus) [inline web/app.js]: This is changing the current behaviour, was that indented?
Are you perhaps trying to fix https://bugzilla.mozilla.org/show_bug.cgi?id=1211884 here as well?

File: web/hand_tool.js [hunk 2]
```diff
@@ -54,52 +47,36 @@ var HandTool = (function HandToolClosure() {
...
-            mozL10n.get('hand_tool_disable.title', null, 'Disable hand tool');
-          this.toggleHandTool.firstElementChild.textContent =
-            mozL10n.get('hand_tool_disable_label', null, 'Disable hand tool');
-        } else {
-          this.toggleHandTool.title =
-            mozL10n.get('hand_tool_enable.title', null, 'Enable hand tool');
-          this.toggleHandTool.firstElementChild.textContent =
-            mozL10n.get('hand_tool_enable_label', null, 'Enable hand tool');
-        }
+        this.eventBus.dispatch('handtoolchanged', {isActive: isActive});
```
Comment (Snuffleupagus) [inline web/hand_tool.js]: Did you mean `..., { isActive: this.isActive });` here?

File: web/secondary_toolbar.js [hunk 3]
```diff
@@ -79,27 +80,46 @@ var SecondaryToolbar = {
         element.addEventListener('click', elements[item].handler.bind(this));
       }
     }
+
+    // Tracking hand tool menu item changes.
```
Comment (Snuffleupagus) [inline web/secondary_toolbar.js]: With these changes, this line can be removed https://github.com/yurydelendik/pdf.js/blob/secondary/web/secondary_toolbar.js#L61.

**Comment by Snuffleupagus** [comment]
Should we change these lines as well: https://github.com/yurydelendik/pdf.js/blob/secondary/web/app.js#L281-L288? Perhaps something like this:
```javascript
  { element: appConfig.fullscreen.contextFirstPage,
    handler: webViewerFirstPage },
  { element: appConfig.fullscreen.contextLastPage,
    handler: webViewerLastPage },
  { element: appConfig.fullscreen.contextPageRotateCw,
    handler: webViewerRotateCw },
  { element: appConfig.fullscreen.contextPageRotateCcw,
    handler: webViewerRotateCcw }
```
```

File: web/app.js (hunk 1 above)
Comment (yurydelendik) [inline web/app.js]: It goes along with changing event handles of toolbar icons to just call eventBus.dispatch (before they called secondary's handlers which were closing itself).

However, change `,false)` => `,true)` below might fix 1211884

File: web/secondary_toolbar.js (hunk 3 above)
Comment (yurydelendik) [inline web/secondary_toolbar.js]: Yes, thank you.

**Comment by yurydelendik** [comment]
> Should we change these lines as well: https://github.com/yurydelendik/pdf.js/blob/secondary/web/app.js#L281-L288? 

The menu items shall generate eventBus command -- changed the code to do just that.

**Comment by Snuffleupagus** [comment]
This line can be removed now too: https://github.com/mozilla/pdf.js/pull/7266/files#diff-529d1853ee1bba753a0fcb40ea778723L275.
Looks good so far, I'll do a final round of testing/reviewing tomorrow.

**Comment by Snuffleupagus** [comment]
Looks good, thank you!

**Pull request merged**